components/automate-cli/pkg/verifyserver/services/batchcheckservice/batchcheckservice.go

```diff
@@ -32,13 +50,255 @@ func (ss *BatchCheckService) BatchCheck(checks []string, config models.Config) m
 		checkTriggerRespMap = getBastionCheckResp(ss, bastionChecks, config)
 	}
 
-	// Get remote check trigger resp and append to checkTriggerRespMap (keys and values)
+	successfullyStartedMockServers, notStartedMockServers, err := ss.startMockServerIfNeeded(remoteChecks, config)
+	if err != nil {
+		ss.log.Error("Error occurred while starting mock server")
+		return models.BatchCheckResponse{}, err
+	}
+
+	if len(notStartedMockServers) > 0 {
+		mockServersStoppedSuccessfully := ss.stopAllMockServers(successfullyStartedMockServers)
+		if !mockServersStoppedSuccessfully {
+			return models.BatchCheckResponse{}, errors.New("mock server stop failed on some nodes")
+		}
+		return models.BatchCheckResponse{}, errors.New("mock server start failed on some nodes")
+	}
 	if len(remoteChecks) > 0 {
 		for key, value := range getRemoteCheckResp(ss, remoteChecks, config) {
 			checkTriggerRespMap[key] = value
 		}
 	}
-	return constructBatchCheckResponse(checkTriggerRespMap, append(bastionChecks, remoteChecks...))
+	mockServersStoppedSuccessfully := ss.stopAllMockServers(successfullyStartedMockServers)
+	if !mockServersStoppedSuccessfully {
+		return models.BatchCheckResponse{}, errors.New("mock server stop failed on some nodes")
+	}
+	return constructBatchCheckResponse(checkTriggerRespMap, append(bastionChecks, remoteChecks...)), nil
+}
+
+func (ss *BatchCheckService) stopAllMockServers(successfullyStartedMockServers []models.MockServerFromBatchServiceResponse) bool {
+	if len(successfullyStartedMockServers) > 0 {
+		stopMockServerChannel := make(chan models.MockServerFromBatchServiceResponse)
```

Comment:
make it buffered since we know length = successfullyStartedMockServers

Reply:
Also close this channel